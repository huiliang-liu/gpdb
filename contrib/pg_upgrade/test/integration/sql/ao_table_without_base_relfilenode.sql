CREATE TABLE ao_without_base_relfilenode (a int ,b int) WITH (appendonly=true);
INSERT INTO ao_without_base_relfilenode VALUES (1,1), (2,2), (3,3);
-- alter the table so that the relfilenod is changed
ALTER TABLE ao_without_base_relfilenode SET DISTRIBUTED RANDOMLY;
-- delete some records so that we have empty base relfilenode files
DELETE FROM ao_without_base_relfilenode;
INSERT INTO ao_without_base_relfilenode VALUES (1,1), (2,2), (3,3);
-- vaccum the table so that the empty base relfilenode files are deleted from system
VACUUM ao_without_base_relfilenode;
-- check that the base relfilenode does not exist on some segments
SELECT pg_stat_file('base/' || db.oid || '/' || pc.relfilenode) from gp_dist_random('pg_class') pc, gp_dist_random('pg_database') db where pc.relname='ao_without_base_relfilenode' and datname = current_database();

CREATE TABLE aoco_without_base_relfilenode WITH (appendonly=true, orientation=column) AS (
  SELECT GENERATE_SERIES::numeric a, GENERATE_SERIES b FROM GENERATE_SERIES(1, 2)
);
UPDATE aoco_without_base_relfilenode SET b=-b;
VACUUM aoco_without_base_relfilenode;
-- check that the base relfilenode does not exist on some segments
SELECT pg_stat_file('base/' || db.oid || '/' || pc.relfilenode) from gp_dist_random('pg_class') pc, gp_dist_random('pg_database') db where pc.relname='aoco_without_base_relfilenode' and datname = current_database();
