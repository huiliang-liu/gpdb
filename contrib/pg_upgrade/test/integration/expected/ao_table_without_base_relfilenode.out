CREATE TABLE ao_without_base_relfilenode (a int ,b int) WITH (appendonly=true);
CREATE
INSERT INTO ao_without_base_relfilenode VALUES (1,1), (2,2), (3,3);
INSERT 3
-- alter the table so that the relfilenod is changed
ALTER TABLE ao_without_base_relfilenode SET DISTRIBUTED RANDOMLY;
ALTER
-- delete some records so that we have empty base relfilenode files
DELETE FROM ao_without_base_relfilenode;
DELETE 3
INSERT INTO ao_without_base_relfilenode VALUES (1,1), (2,2), (3,3);
INSERT 3
-- vaccum the table so that the empty base relfilenode files are deleted from system
VACUUM ao_without_base_relfilenode;
VACUUM
-- check that the base relfilenode does not exist on some segments
SELECT pg_stat_file('base/' || db.oid || '/' || pc.relfilenode) from gp_dist_random('pg_class') pc, gp_dist_random('pg_database') db where pc.relname='ao_without_base_relfilenode' and datname = current_database();
ERROR:  could not stat file "base/16384/16388": No such file or directory  (seg0 slice2 192.168.1.72:50002 pid=24747)

CREATE TABLE aoco_without_base_relfilenode WITH (appendonly=true, orientation=column) AS ( SELECT GENERATE_SERIES::numeric a, GENERATE_SERIES b FROM GENERATE_SERIES(1, 2) );
CREATE 2
UPDATE aoco_without_base_relfilenode SET b=-b;
UPDATE 2
VACUUM aoco_without_base_relfilenode;
VACUUM
-- check that the base relfilenode does not exist on some segments
SELECT pg_stat_file('base/' || db.oid || '/' || pc.relfilenode) from gp_dist_random('pg_class') pc, gp_dist_random('pg_database') db where pc.relname='aoco_without_base_relfilenode' and datname = current_database();
ERROR:  could not stat file "base/16384/16392": No such file or directory  (seg0 slice2 192.168.1.72:50002 pid=24747)
